pkg/mount/gpt/gpt.go

```diff
@@ -41,19 +41,20 @@ type (
 	MBR    [BlockSize]byte
 	Header struct {
 		Signature  uint64
-		Revision   uint32 // (for GPT version 1.0 (through at least UEFI version 2.7 (May 2017)), the value is 00h 00h 01h 00h)
-		HeaderSize uint32 // size in little endian (in bytes, usually 5Ch 00h 00h 00h or 92 bytes)
-		CRC        uint32 // CRC32/zlib of header (offset +0 up to header size) in little endian, with this field zeroed during calculation
-		Reserved   uint32 // ; must be zero
-		CurrentLBA uint64 // (location of this header copy)
-		BackupLBA  uint64 // (location of the other header copy)
-		FirstLBA   uint64 // usable LBA for partitions (primary partition table last LBA + 1)
-		LastLBA    uint64 // usable LBA (secondary partition table first LBA - 1)
-		DiskGUID   GUID   // (also referred as UUID on UNIXes)
-		PartStart  uint64 // LBA of array of partition entries (always 2 in primary copy)
-		NPart      uint32 // Number of partition entries in array
-		PartSize   uint32 // Size of a single partition entry (usually 80h or 128)
-		PartCRC    uint32 // CRC32/zlib of partition array in little endian
+		Revision   uint32    // (for GPT version 1.0 (through at least UEFI version 2.7 (May 2017)), the value is 00h 00h 01h 00h)
+		HeaderSize uint32    // size in little endian (in bytes, usually 5Ch 00h 00h 00h or 92 bytes)
+		CRC        uint32    // CRC32/zlib of header (offset +0 up to header size) in little endian, with this field zeroed during calculation
+		Reserved   uint32    // ; must be zero
+		CurrentLBA uint64    // (location of this header copy)
+		BackupLBA  uint64    // (location of the other header copy)
+		FirstLBA   uint64    // usable LBA for partitions (primary partition table last LBA + 1)
+		LastLBA    uint64    // usable LBA (secondary partition table first LBA - 1)
+		DiskGUID   GUID      // (also referred as UUID on UNIXes)
+		PartStart  uint64    // LBA of array of partition entries (always 2 in primary copy)
+		NPart      uint32    // Number of partition entries in array
+		PartSize   uint32    // Size of a single partition entry (usually 80h or 128)
+		PartCRC    uint32    // CRC32/zlib of partition array in little endian
+		_          [420]byte // Reserved; must be zero.
```

Comment:
The only problem is that this assumes a sector of size 512 bytes. But if that is safe for our use cases, I don't see any problem with it.

Original code only write allocated fields, the first 92 bytes, and assumes rest of the block is zero-ed. That will work with any sector size. But we cant be sure of the default values of rest of the block. So explicitly zero-ing it out is advantageous.

Reply:
Given we define block size in `BlockSize`, perhaps, this should be `_  [BlockSize-0x5C]byte` ?

Reply:
OR `_ [BlockSize - HeaderSize]byte`

Reply:
Changed to "_ [BlockSize - HeaderSize]byte" as suggested

Regarding the 512-byte assumption and safety of zeroing these blocks, there's an ongoing conversation in the pull request so I'll leave this unresolved until we converge on a conclusion there :)

Reply:
Resolving conversation as this code is no longer part of the proposed pull.